mintpy/prep_fringe.py

```diff
@@ -0,0 +1,242 @@
+    print("-"*50)
+    print("create HDF5 file {} with w mode".format(fname))
+    h5 = h5py.File(fname, "w")
+
+    # initiate dataset
+    for key in dsNameDict.keys():
+        compression = None
+
+        # changable dataset shape
+        if len(dsNameDict[key][1]) == 3:
+            maxShape = (None, dsNameDict[key][1][1], dsNameDict[key][1][2])
+        else:
+            maxShape = dsNameDict[key][1]
+
+        print("create dataset: {d:<25} of {t:<25} in size of {s}".format(
+            d=key,
+            t=str(dsNameDict[key][0]),
+            s=dsNameDict[key][1]))
+        h5.create_dataset(key,
+                          shape=dsNameDict[key][1],
+                          maxshape=maxShape,
+                          dtype=dsNameDict[key][0],
+                          chunks=True,
+                          compression=compression)
+
+    # write attributes
+    for key in metadata.keys():
+        h5.attrs[key] = metadata[key]
+
+    h5.close()
+    print("close HDF5 file {}".format(fname))
+    return
+
+
+def write_geometry(outFile, geomDir, bbox):
+    f = h5py.File(outFile, "a")
+    
+    hgt = os.path.join(geomDir, "hgt.rdr.full")
+    lat = os.path.join(geomDir, "lat.rdr.full")
+    lon = os.path.join(geomDir, "lon.rdr.full")
+    los = os.path.join(geomDir, "los.rdr.full")
+    shdw = os.path.join(geomDir, "shadowMask.rdr.full")
+    
+    geomDict = {"height":hgt,
+                "latitude":lat, 
+                "longitude":lon,
+                "incidenceAngle":los,
+                "azimuthAngle":los,
+                "shadowMask":shdw}
+    
+    for key in geomDict:
+        ds = gdal.Open(geomDict[key], gdal.GA_ReadOnly)
+
+        if key == "ShadowMaxk":
+            data = np.array(ds.ReadAsArray(), dtype=np.byte)
+        else:
+            data = np.array(ds.ReadAsArray(), dtype=np.float32)
+        
+        data[data == ds.GetRasterBand(1).GetNoDataValue()] = np.nan 
+
+        if key == "incidenceAngle":
+            data = data[0]
+        elif key == "azimuthAngle":
+            data = data[1]
+        
+        f[key][:,:] = data[bbox[1]:bbox[3], bbox[0]:bbox[2]]
+    
+    meta = dict(f.attrs)
+    meta["LENGTH"], meta["WIDTH"] = data.shape
+    slantRangeFull = utils.range_distance(meta, dimension=2)
+    f["slantRangeDistance"][:,:] = slantRangeFull[bbox[1]:bbox[3], bbox[0]:bbox[2]]
+    f.close()
+
+    return outFile
+
+
+def write_timeseries(outFile, ifgs, baselineDir):
+    #Prepare timeseries data
+    f = h5py.File(outFile, "a")
+    
+    bperpDict = read_baseline_timeseries(baselineDir, processor="tops")
+    
+    dates = list(bperpDict.keys())
+    dates.sort()
+    f["date"][:,] = [np.string_(x) for x in dates]
+    
+    bperps = [bperpDict[x][0] for x in dates]
+    f["bperp"][:,] = bperps
+
+    #Add data to arrays
+    for i in range(0, len(ifgs)):
+        ifg = gdal.Open(ifgs[i])
+        array = np.array(ifg.GetRasterBand(2).ReadAsArray())
+        f["timeseries"][i+1] = array
+
```

Comment:
The `layout_hdf5()` just initiate the h5 dataset object. The time-series data for the reference date is zero, my first guess is that it still needs to be explicitly written to disk. Could you please confirm it? @4restwilliams

Reply:
![Figure_1_timeseries](https://user-images.githubusercontent.com/31411324/80164714-d5646c00-862d-11ea-88b0-d030cb438fae.png)

Based on my test data it looks we don't need to, but we can set it explicitly

Reply:
I've added `f["timeseries"][0] = np.zeros_like(array, dtype=np.float32)` to enforce this.